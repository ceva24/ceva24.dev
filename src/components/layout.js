import React from "react"
import { css } from "@emotion/core"
import { rhythm } from "src/utils/typography"
import Helmet from "react-helmet"
import GlobalStyles from "src/styles/global"
import Bio from "src/components/layout/bio"

export default ({ title, children }) => (
    <div css={css`
        margin: 0 auto;
        min-width: 300px;
        max-width: 800px;
        padding: ${rhythm(3 / 2)} ${rhythm(1)};
    `}>

        <GlobalStyles />

        <Helmet>
            <html lang="en" />
            <meta charSet="utf-8" />
            <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
            <title>{title}</title>
        </Helmet>

        <Bio />

        {children}
    </div>
)
