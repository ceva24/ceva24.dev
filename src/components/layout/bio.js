import React from "react"
import { css } from "@emotion/core"
import { faEnvelope } from "@fortawesome/free-solid-svg-icons"
import { faGithub, faLinkedin, faTwitter } from "@fortawesome/free-brands-svg-icons"
import { rhythm } from "src/utils/typography"
import SocialIcon from "src/components/socialIcon"
import Avatar from "static/avatar.png"

export default () => (

    <div css={css`
            display: flex;
            align-items: center;
            margin-bottom: ${rhythm(2)};
    `}>
        <img
            src={Avatar}
            alt="Chris Evans"
            css={css`
                margin-right: ${rhythm(1 / 2)};
                margin-bottom: 0;
                border-radius: 100%;
                height: ${rhythm(3)};
                width: ${rhythm(3)};
        `} />
        <div css={css`
            padding-left: ${rhythm(3 / 4)};
            border-left: 1px solid;
        `}
        >
            <strong>Chris Evans</strong>
            <div>A Web Development Team Leader at the University of York</div>
            <div css={css`
                font-size: ${rhythm(4 / 5)};
            `}>
                <SocialIcon icon={faEnvelope} url="mailto:chris@ceva24.dev" />
                <SocialIcon icon={faGithub} url="https://www.github.com/ceva24" />
                <SocialIcon icon={faLinkedin} url="https://uk.linkedin.com/in/ceva24" />
                <SocialIcon icon={faTwitter} url="https://twitter.com/ceva24" />
            </div>
        </div>
    </div>
)
