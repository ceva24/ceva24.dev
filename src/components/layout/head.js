import React from "react"
import Helmet from "react-helmet"

export default (props) => (
    <Helmet>
        <title>{props.title} | {props.subtitle}</title>
    </Helmet>
)
