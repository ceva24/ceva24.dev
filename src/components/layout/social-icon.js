import React from "react"
import { css } from "@emotion/core"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { rhythm } from "src/utils/typography"

export default (props) => (
    <a href={props.url} title={props.url} aria-label={props.label} css={css`
        color: inherit;
        margin-right: ${rhythm(1 / 2)};
    `}>
        <FontAwesomeIcon icon={props.icon} />
    </a>
)
