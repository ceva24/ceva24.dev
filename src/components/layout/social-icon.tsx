import { css } from "@emotion/react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { IconProp } from "@fortawesome/fontawesome-svg-core";
import { rhythm } from "../../utils/typography";

type SocialIconProps = {
    url: string;
    label: string;
    icon: IconProp;
};

const SocialIcon = ({ url, label, icon }: SocialIconProps) => (
    <a
        href={url}
        title={url}
        aria-label={label}
        css={css`
            color: inherit;
            margin-right: ${rhythm(1 / 2)};
        `}
    >
        <FontAwesomeIcon icon={icon} />
    </a>
);

export {SocialIcon };
