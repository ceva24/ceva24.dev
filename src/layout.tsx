import { css } from "@emotion/react";
import Helmet from "react-helmet";
import { rhythm } from "./styles/typography";
import { GlobalStyles } from "./styles/global";
import { Bio } from "./components/bio";

interface LayoutProps {
    title: string;
    children: React.ReactNode;
}

const Layout: React.FC<LayoutProps> = ({ title, children }: LayoutProps) => (
    <div
        css={css`
            margin: 0 auto;
            min-width: 300px;
            max-width: 800px;
            padding: ${rhythm(3 / 2)} ${rhythm(1)};
        `}
    >
        <GlobalStyles />

        <Helmet>
            <html lang="en" />
            <meta charSet="utf-8" />
            <meta
                name="viewport"
                content="width=device-width, initial-scale=1, shrink-to-fit=no"
            />
            <meta
                name="description"
                content="Chris Evans, a Web Development / Systems Integration Team Leader at the University of York"
            />
            <title>{title}</title>
        </Helmet>

        <Bio />

        {children}
    </div>
);

export { Layout };
