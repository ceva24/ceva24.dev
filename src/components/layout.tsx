import React from "react";
import Helmet from "react-helmet";
import { graphql, useStaticQuery, Link } from "gatsby";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import {
    faGithub,
    faLinkedin,
    faTwitter,
} from "@fortawesome/free-brands-svg-icons";
import { SocialIcon } from "./social-icon";

interface LayoutProps {
    title?: string;
    children: React.ReactNode;
}

interface PureLayoutProps extends LayoutProps {
    pageDescription: string;
    name: string;
    website: string;
    subtitle: string;
}

const PureLayout: React.FC<PureLayoutProps> = ({
    title,
    name,
    website,
    subtitle,
    pageDescription,
    children,
}: PureLayoutProps) => {
    const pageTitle = title ?? name;

    return (
        <div className="text-lg font-light">
            <Helmet>
                <html lang="en" />
                <meta charSet="utf-8" />
                <meta
                    name="viewport"
                    content="width=device-width, initial-scale=1, shrink-to-fit=no"
                />
                <meta name="description" content={pageDescription} />
                <title>{pageTitle}</title>
            </Helmet>

            <header className="max-w-6xl mx-auto border-b">
                <div className="my-8 text-center space-y-5">
                    <h1 className="text-7xl">
                        <Link to="/">{name}</Link>
                    </h1>

                    <div>
                        <strong>{website}</strong> | {subtitle}
                    </div>

                    <div
                        className="space-x-8 text-gray-700 text-2xl"
                        aria-label="Contact"
                    >
                        <SocialIcon
                            icon={faEnvelope}
                            label="Email"
                            url="mailto:chris@ceva24.dev"
                        />
                        <SocialIcon
                            icon={faGithub}
                            label="Github"
                            url="https://www.github.com/ceva24"
                        />
                        <SocialIcon
                            icon={faLinkedin}
                            label="LinkedIn"
                            url="https://www.linkedin.com/in/ceva24"
                        />
                        <SocialIcon
                            icon={faTwitter}
                            label="Twitter"
                            url="https://twitter.com/ceva24"
                        />
                    </div>
                </div>
            </header>

            <main className="max-w-6xl mx-auto px-3 my-6">{children}</main>
        </div>
    );
};

const Layout: React.FC<LayoutProps> = ({ title, children }: LayoutProps) => {
    const data: LayoutData = useStaticQuery(
        // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
        graphql`
            query {
                site {
                    siteMetadata {
                        name
                        website
                        subtitle
                        pageDescription
                    }
                }
            }
        `
    );

    return (
        <PureLayout
            title={title}
            name={data.site.siteMetadata.name}
            website={data.site.siteMetadata.website}
            subtitle={data.site.siteMetadata.subtitle}
            pageDescription={data.site.siteMetadata.pageDescription}
        >
            {children}
        </PureLayout>
    );
};

export { Layout, PureLayout };
