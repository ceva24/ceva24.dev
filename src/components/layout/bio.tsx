import { graphql, useStaticQuery, Link } from "gatsby";
import { css } from "@emotion/react";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import {
    faGithub,
    faLinkedin,
    faTwitter,
} from "@fortawesome/free-brands-svg-icons";
import { rhythm } from "../../utils/typography";
import ProfilePicture from "../../../static/profile-picture.png";
import { SocialIcon } from "./social-icon";

const Bio = () => {
    const data: BioComponentData = useStaticQuery(
        // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
        graphql`
            query {
                site {
                    siteMetadata {
                        name
                        subtitle
                    }
                }
            }
        `
    );

    return (
        <div
            css={css`
                display: flex;
                align-items: center;
                margin-bottom: ${rhythm(2)};
                @media (max-width: 768px) {
                    justify-content: center;
                }
            `}
        >
            <img
                src={ProfilePicture as string}
                alt="Chris Evans head shot"
                css={css`
                    margin-right: ${rhythm(1 / 2)};
                    margin-bottom: 0;
                    border-radius: 100%;
                    min-height: 78px;
                    min-width: 78px;
                    height: 78px;
                    width: 78px;
                    box-shadow: none;
                `}
            />
            <div
                css={css`
                    padding-left: ${rhythm(3 / 4)};
                    border-left: 1px solid;
                `}
            >
                <h1>
                    <Link
                        to="/"
                        className="bio-link"
                        css={css`
                            color: inherit;
                            &:hover {
                                color: inherit;
                            }
                        `}
                    >
                        {data.site.siteMetadata.name}
                    </Link>
                </h1>
                <div>
                    A Web Development / Systems Integration Team Leader at the
                    University of York
                </div>
                <div
                    css={css`
                        @media (max-width: 768px) {
                            font-size: ${rhythm(1)};
                        }
                        @media (min-width: 768px) {
                            font-size: ${rhythm(4 / 5)};
                        }
                    `}
                >
                    <SocialIcon
                        icon={faEnvelope}
                        label="Email"
                        url="mailto:chris@ceva24.dev"
                    />
                    <SocialIcon
                        icon={faGithub}
                        label="Github"
                        url="https://www.github.com/ceva24"
                    />
                    <SocialIcon
                        icon={faLinkedin}
                        label="LinkedIn"
                        url="https://uk.linkedin.com/in/ceva24"
                    />
                    <SocialIcon
                        icon={faTwitter}
                        label="Twitter"
                        url="https://twitter.com/ceva24"
                    />
                </div>
            </div>
        </div>
    );
};

export { Bio };
