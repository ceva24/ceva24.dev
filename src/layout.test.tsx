import React from "react";
import { render } from "@testing-library/react";
import { PureLayout } from "./layout";

describe("layout", () => {
    it("renders", () => {
        const bio = <div>Bio</div>;
        const children = <p>Hello</p>;

        const { asFragment } = render(
            <PureLayout title="Title" description="Description" bio={bio}>
                {children}
            </PureLayout>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
