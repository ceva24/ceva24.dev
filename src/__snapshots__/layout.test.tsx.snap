// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      Bio
    </div>
    <p>
      Hello
    </p>
  </div>
</DocumentFragment>
`;
